Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -66,11 +71,11 @@ public class NotificationPublisher : INotificationPublisher {
     }
 
     private void PublishGlobalProgress(IdeState previousState, IdeState state) {
-      foreach (var uri in state.Compilation.RootUris) {
+      foreach (var uri in state.Input.RootAndProjectUris) {
```

Comment:
Do you know what the TODO below is about?

Reply:
Yes. Each file is owned by only one project, and files shouldn't publish notifications for files that they don't own. That ownership check is missing here, so one file could be "Parsing..." in one project and "Resolving..." and another, and both projects could publish competing notifications about that.